### DISCUSSION TITLE
Mixing arm64 and amd64: ... has incorrect configuration: Expected MinIO binary checksum...

### DISCUSSION THREAD

**Discussion by docbobo** [minio/minio#20605] [Q&A] [answered]
So, I was just adding a node with a different processor architecture to my k8s cluster and when MinIO came up on that node, it was reporting

```
 Following servers have mismatching configuration [...] has incorrect configuration: Expected MinIO binary checksum: [...]
```

Does MinIO require all instances to be running on the same processor architecture?

**Comment by harshavardhana** [comment, marked as answer]
> Does MinIO require all instances to be running on the same processor architecture?

Of course.

**Comment by JKaique01** [comment]
Hello,

although you probably shouldn't be mixing architectures...

I just wanted to share how I managed to get it working via QEMU for docker.
The current setup seems to be working just fine, please let me know if there something I should be concerned of.

---

Here is how I achieved it:

[Install QEMU for docker](https://docs.docker.com/build/building/multi-platform/#install-qemu-manually)

then put `platform: linux/arm64` at my docker compose service (in the amd64 machines)

---

I first tried to run the amd64 image at my arm machine, but got `minio  | Fatal glibc error: CPU does not support x86-64-v2` because my arm64 processor didnt match the requirements for emulation in this case, but then the amd64 machines could emulate the arm64.

I also noticed that if you run just two machines (will be just a replication), it minio wont complain about the missmatch, but as soon as you get at least three machines, ensure coding will kick in and require same architectures.
